tests/media_filename_test.go

```diff
@@ -179,17 +298,25 @@ func TestMediaFilenames(t *testing.T) {
 }
 
 // Returns content disposition information like (filename, isAttachment)
-func downloadForFilename(t *testing.T, c *client.CSAPI, mxcUri string, diffName string) (filename string, isAttachment bool) {
+func downloadForFilename(t *testing.T, c *client.CSAPI, mxcUri string, diffName string, newPath bool) (filename string, isAttachment bool) {
 	t.Helper()
 
 	origin, mediaId := client.SplitMxc(mxcUri)
 
 	var path []string
 
 	if diffName != "" {
-		path = []string{"_matrix", "media", "v3", "download", origin, mediaId, diffName}
+		if newPath {
+			path = []string{"_matrix", "client", "v1", "media", "download", origin, mediaId, diffName}
+		} else {
+			path = []string{"_matrix", "media", "v3", "download", origin, mediaId, diffName}
+		}
```

Comment:
```suggestion
        path = []string{"_matrix", "media", "v3", "download", origin, mediaId, diffName}
		if newPath {
			path = []string{"_matrix", "client", "v1", "media", "download", origin, mediaId, diffName}
		} 
```
imo easier to read. Same for the else block below.